src/View/AjaxView.php

```diff
@@ -122,12 +123,18 @@ public function render($view = null, $layout = null) {
 	 *
 	 * @param array $dataToSerialize Array of data that is to be serialzed.
 	 * @return string The serialized data.
+	 * @throws \RuntimeException
 	 */
-	protected function _serialize($dataToSerialize = []) {
+	protected function _serialize(array $dataToSerialize = []) {
 		$result = json_encode($dataToSerialize);
 		if (json_last_error() !== JSON_ERROR_NONE) {
-			return json_encode(['error' => json_last_error_msg()]);
+			$result = json_encode(['error' => json_last_error_msg()]);
```

Comment:
Are we sure we want to leak this kind of info to a user? Shouldnt this rather be logged, and a normal user error returned?